security/identity-and-access-management/oracle-identity-governance/webservice-samples/extend-access-ws/files/src/com/oracle/samplews/ExtendAccessWSv1.java

```diff
@@ -0,0 +1,348 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *******************************************************************************/
+
+package com.oracle.samplews;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.security.auth.login.LoginException;
+import javax.sql.DataSource;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import oracle.iam.platform.OIMClient;
+import oracle.iam.platform.entitymgr.vo.SearchCriteria;
+import oracle.iam.provisioning.api.ProvisioningConstants;
+import oracle.iam.provisioning.api.ProvisioningService;
+import oracle.iam.provisioning.vo.Account;
+import oracle.iam.provisioning.vo.EntitlementInstance;
+
+/**
+ * Sample RESTful web service implementation that extends a user's access
+ * following a REST call with the correct access extension UUID.
+ * 
+ * Must be used in conjunction with the AccessTerminationNotification scheduled
+ * task, which will generate the required database entries for server-side
+ * validation and send access extension links via email.
+ * 
+ * @author mgere-oracle
+ *
+ */
+@Path("/v1")
+public class ExtendAccessWSv1 {
+	private static final Logger logger = Logger.getLogger(ExtendAccessWSv1.class.getName());
+
+	private static final String EXTENSION_SELECT_QUERY = "SELECT EXT_INSTANCE_TYPE, EXT_ACCESS_ID, EXT_INSTANCE_NAME, EXT_ACCESS_END_DATE FROM EXTEND_ACCESS WHERE EXT_UUID = ? AND EXT_USR_KEY = ?";
+	private static final String EXTENSION_DELETE_QUERY = "DELETE FROM EXTEND_ACCESS WHERE EXT_UUID = ?";
+
+	private OIMClient oimClient;
+
+	/**
+	 * Connect to the remote/local OIM node using an OIMClient object
+	 * 
+	 * @throws IOException
+	 * @throws LoginException
+	 */
+	private void initOIMClient() throws IOException, LoginException {
+		logger.log(Level.FINE, "Initializing OIM Client...");
+
+		Properties prop = new Properties();
+		InputStream configurationsFileInput = null;
+
+		String wlDomainDirectory = System.getProperty("domain.home");
+		configurationsFileInput = new FileInputStream(
+				java.nio.file.Paths.get(wlDomainDirectory, "config", "extendaccessws.properties").toString());
```

Comment:
Hidden imports in the code are not a good coding practice. Add this class to the import section.

Reply:
Ouch indeed. Not sure how my linter let this through. Fixed.